providers/facebook/facebook.go

```diff
@@ -37,6 +37,7 @@ func New(clientKey, secret, callbackURL string, scopes ...string) *Provider {
 		providerName: "facebook",
 	}
 	p.config = newConfig(p, scopes)
+	p.Fields = "email,first_name,last_name,link,about,id,name,picture,location"
```

Comment:
In the interest of avoiding any more issues with the Facebook provider in the future, I'm wondering, what is the relationship between a "scope" and a "field" when it comes to permissions? Do the scopes match one-to-one with the fields? For example, if I authenticate with the scope "email", am I allowed to see the user's name and location, or would I need additional scopes for that?

Reply:
Yeah, that makes us confused. Parameters of scopes are not same with those of fields. As you know, "scope" is used for user permissions, what type of information is permitted to access by the token. Easiest way to try to find which permissions are available is at [Graph API Explorer](https://developers.facebook.com/tools/explorer/).
On the other hand, "fields" is for which actual field you want to get, post, delete.

Here are some examples.
- When I get a token with scope "email", it can get "id", "name", "email". "id" and "name" are default parameters in Graph api v3.2.
- If I want to get users' profile links, I have to set "user_link" in scopes. Then I can retrieve the link by setting fields="link".

Reply:
Ok, thanks, I think I get it. Scopes manage permissions, and the fields work in a sort of GraphQL way -- you can request specific data via the scopes.

I'm happy to merge this, hopefully this resolves the issues. Thanks for the contribution @kobayashi!